plugins/out_kinesis_streams/kinesis_api.c

```diff
@@ -97,6 +97,31 @@ static int init_put_payload(struct flb_kinesis *ctx, struct flush *buf,
     return -1;
 }
 
+/*
+ * Adjust the time_key output format according to time_key_format.
+ */
+static void get_time_format_string(char *str, size_t maxsize,
+                      const char *time_key_format, const struct tm *timeptr,
+                      size_t len, struct flb_time *tms){
+    size_t s;
+    const char *format = "%Y-%m-%dT%H:%M:%S";
+    len = strftime(str, maxsize, format, timeptr);
+    format = strstr(time_key_format, ".%F");
+    if (format) {
+        s = snprintf(str + len, sizeof(str) - 1 - len,
+                     ".%03" PRIu64 "Z",
+                     (uint64_t) tms->tm.tv_nsec / 1000000);
+    } else {
+        format = strstr(time_key_format, ".%L");
+        if (format) {
+            s = snprintf(str + len, sizeof(str) - 1 - len,
+                         ".%09" PRIu64 "Z",
```

Comment:
what does the format var do?

Reply:
because you said we can't put function as argument so I reused the format to avoid that happen

Reply:
Sorry I am not following here- format is a char *, not a function pointer.

`format` is set to the return value of strstr but then its not used later in the function... is the goal just to check whether or not the format contains this string? What is the purpose of this variable?